src/runtime/audience-action-local-flow.ts

```diff
@@ -115,107 +119,115 @@ export class AudienceActionLocalFlow implements AudienceActionFlow {
 
     const shadow = wrapper.attachShadow({mode: 'open'});
 
-    shadow.appendChild(prompt);
+    shadow.appendChild(this.prompt_);
 
     return wrapper;
   }
 
-  private renderErrorView_(): HTMLElement {
-    const prompt = createElement(this.doc_, 'div', {});
-    return this.makeErrorView(prompt);
-  }
-
-  private makeErrorView(prompt: HTMLElement): HTMLElement {
-    prompt./*OK*/ innerHTML = ERROR_HTML;
-    return prompt;
+  private renderErrorView_() {
+    // TODO: mhkawano - Make closeable.
+    // TODO: mhkawano - Make look nicer.
+    this.prompt_./*OK*/ innerHTML = ERROR_HTML;
   }
 
-  private renderAndInitRewardedAdWall_(): HTMLElement {
-    // Setup callback for googletag init.
-    const googletag = this.deps_.win().googletag;
-    googletag.cmd.push(this.initRewardedAdWall_.bind(this));
-
-    const prompt = createElement(this.doc_, 'div', {});
-    setImportantStyles(prompt, {
+  private renderLoadingView_() {
+    setImportantStyles(this.prompt_, {
       'height': '100%',
       'display': 'flex',
       'display-flex-direction': 'column',
     });
-
-    prompt./*OK*/ innerHTML = LOADING_HTML;
-
-    return prompt;
+    this.prompt_./*OK*/ innerHTML = LOADING_HTML;
   }
 
-  private async initRewardedAdWall_() {
-    // TODO: mhkawano - Get action config.
-    // TODO: mhkawano - support premon.
-
-    // Init gpt.js
-    const initGptPromise = new Promise<boolean>(this.initGpt_.bind(this));
-    const initSuccess = await initGptPromise;
-
-    // Replace with error view if init fails.
-    // TODO: mhkawano - Make closeable.
-    if (!initSuccess) {
-      this.makeErrorView(this.prompt_);
+  private async initPrompt_() {
+    if (this.params_.action === TYPE_REWARDED_AD) {
+      const config = await this.getConfig_();
+      // Setup callback for googletag init.
+      const googletag = this.deps_.win().googletag;
+      googletag.cmd.push(() => {
```

Comment:
How would the prompt look like if this callback execution is delayed? Should we handle the case when it's never executed?

Reply:
Moved the timeout up a level so that it should catch the case where it's not called.